### PR TITLE
acceptance test for creating guest with email containing capital letter

### PR DESCRIPTION
**Pull request by HariBhandari07**
## Description

acceptance test for creating guest with email containing capital letter
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by HariBhandari07** [comment]
on top of https://github.com/owncloud/guests/pull/327

**Comment by HariBhandari07** [comment]
On top of #327

File: tests/acceptance/features/bootstrap/WebUIGuestsContext.php [hunk 1]
```diff
@@ -91,12 +129,57 @@ public function guestUserRegistersUsingWebUI($guestDisplayName, $password) {
...
+		$this->filesPage->openSharingDialog($fileName, $this->getSession());
+		$sharingDialog = $this->filesPage->getSharingDialog();
+		$sharingDialog->shareWithUserOrGroup(
+			$email, $this->preUserAddDialog . $email . $this->postUserAddDialog, $this->getSession()
+		);
+		$this->featureContext->addUserToCreatedUsersList($email, null);
+	}
+
+	/**
+	 * @Then user should not be able to share the file with guest user :user
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/WebUIGuestsContext.php]:
```suggestion
	 * @Then the user should not be able to share the file with guest user :user
```

File: tests/acceptance/features/bootstrap/WebUIGuestsContext.php [hunk 2]
```diff
@@ -91,12 +129,57 @@ public function guestUserRegistersUsingWebUI($guestDisplayName, $password) {
...
+	 * @param string $fileName
+	 * @param string $email
+	 *
+	 * @return void
+	 */
+	public function theUserSharesFileWithGuestUserWithEmailUsingWebui($fileName, $email) {
+		$this->filesPage->openSharingDialog($fileName, $this->getSession());
+		$sharingDialog = $this->filesPage->getSharingDialog();
+		$sharingDialog->shareWithUserOrGroup(
+			$email, $this->preUserAddDialog . $email . $this->postUserAddDialog, $this->getSession()
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/WebUIGuestsContext.php]: Just a suggestion. We used a template in `activity` app to generate strings, like 

https://github.com/owncloud/activity/blob/294b8b745f6ba1178318c1c3eca2477075e27056/tests/acceptance/features/bootstrap/WebUIActivityContext.php#L113

If this is used for creating strings out of template, that might make it more readable.

File: tests/acceptance/features/bootstrap/WebUIGuestsContext.php [hunk 3]
```diff
@@ -55,21 +58,56 @@ class WebUIGuestsContext extends RawMinkContext implements Context {
...
+	 * @param FilesPage $filesPage
 	 */
-	public function __construct(SetPasswordPage $setPasswordPage) {
+	public function __construct(SetPasswordPage $setPasswordPage, FilesPage $filesPage) {
 		$this->setPasswordPage = $setPasswordPage;
+		$this->filesPage = $filesPage;
+	}
+
+	/**
+	 * @return |null
```
Comment (skshetry) [inline tests/acceptance/features/bootstrap/WebUIGuestsContext.php]:
```suggestion
	 * @return string|null
```

File: tests/acceptance/features/bootstrap/WebUIGuestsContext.php [hunk 4]
```diff
@@ -91,12 +129,57 @@ public function guestUserRegistersUsingWebUI($guestDisplayName, $password) {
...
+	 * @Then user should not be able to share the file with guest user :user
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 */
+	public function userShouldNotBeDisplayedInTheDropdownAsGuestUser($user) {
+		$sharingDialogue = $this->filesPage->getSharingDialog();
+		$arrayList = $sharingDialogue->getAutoCompleteItemsList();
+		PHPUnit\Framework\Assert::assertNotContains($this->preUserAddDialog . $user . $this->postUserAddDialog, $arrayList);
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/WebUIGuestsContext.php]: please make that line shorter
```suggestion
		PHPUnit\Framework\Assert::assertNotContains(
		    $this->preUserAddDialog . $user . $this->postUserAddDialog, $arrayList
		);
```

File: tests/acceptance/features/bootstrap/WebUIGuestsContext.php [hunk 5]
```diff
@@ -91,12 +129,57 @@ public function guestUserRegistersUsingWebUI($guestDisplayName, $password) {
...
+		$this->featureContext->addUserToGroup($user, $this->getGuestGroupName());
+	}
+
 	/**
 	 * @Then a warning should be displayed on the set-password-page saying :expectedMessage
 	 *
 	 * @param string $expectedMessage
 	 *
-	 * @return void
+	 * @return boolean/void
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/WebUIGuestsContext.php]:
```suggestion
	 * @return boolean|void
```

File: tests/acceptance/features/bootstrap/WebUIGuestsContext.php [hunk 6]
```diff
@@ -25,6 +25,9 @@
 use Behat\MinkExtension\Context\RawMinkContext;
 use Page\SetPasswordPage;
 use PHPUnit\Framework\Assert;
+use SensioLabs\Behat\PageObjectExtension\PageObject\Exception\ElementNotFoundException;
+use WebDriver\Key;
```
Comment (paurakhsharma) [inline tests/acceptance/features/bootstrap/WebUIGuestsContext.php]: I don't see this being used anywhere. Please remove this if not required.

File: tests/acceptance/features/bootstrap/WebUIGuestsContext.php [hunk 7]
```diff
@@ -127,5 +210,6 @@ public function before(BeforeScenarioScope $scope) {
 		$this->featureContext = $environment->getContext('FeatureContext');
 		$this->webUIGeneralContext = $environment->getContext('WebUIGeneralContext');
 		$this->guestsContext = $environment->getContext('GuestsContext');
+		$this->sharingContext = $environment->getContext('WebUISharingContext');
```
Comment (paurakhsharma) [inline tests/acceptance/features/bootstrap/WebUIGuestsContext.php]: I don't see this being used anywhere. Please remove this if not required.

File: tests/acceptance/config/behat.yml [hunk 8]
```diff
@@ -26,7 +26,11 @@ default:
         - WebUIFilesContext:
         - WebUILoginContext:
         - WebUIGuestsContext:
+        - WebUISharingContext:
+        - WebUIUsersContext:
+        - OccContext:
         - FeatureContext: *common_feature_context_params
+
```
Comment (skshetry) [inline tests/acceptance/config/behat.yml]: Chop this line

File: tests/acceptance/features/webUIGuests/guests.feature [hunk 9]
```diff
@@ -110,3 +110,22 @@ Feature: Guests
...
     Then user "user0" should not exist
+
+  @mailhog
+  @issue-222
+  Scenario Outline: User creates a guest user with email that contains capital letters
+    Given user "user0" has been created with default attributes
+    And user "user0" has logged in using the webUI
+    When the user shares file "data.zip" with guest user with email "<share-email>" using webUI
+    And the user logs out of the webUI
+    When guest user "<share-email>" registers with email "<register-email>" and sets password to "password" using the webUI
```
Comment (dpakach) [inline tests/acceptance/features/webUIGuests/guests.feature]:
```suggestion
    And guest user "<share-email>" registers with email "<register-email>" and sets password to "password" using the webUI
```

File: tests/acceptance/features/webUIGuests/guests.feature [hunk 10]
```diff
@@ -110,3 +110,22 @@ Feature: Guests
     And the administrator has browsed to the users page
     When the administrator deletes user "user0" using the webUI and confirms the deletion using the webUI
     Then user "user0" should not exist
+
+  @mailhog
+  @issue-222
```
Comment (skshetry) [inline tests/acceptance/features/webUIGuests/guests.feature]: Please write this with above line

**Comment by individual-it** [comment]
@HariBhandari07 please resolve the conflicts, hopefully we can merge that then

**Comment by HariBhandari07** [comment]
@individual-it the conflicts have been resolved

**Pull request merged**